### PR TITLE
http3: correctly use the quic.Transport

### LINKED ISSUE #3868: http3: incorrect use of the quic.Transport
@kgersen reports in https://github.com/quic-go/quic-go/pull/3867#issuecomment-1570038847 that the http3 package is not correctly reusing connections, leading to panics in the multiplexer: https://github.com/quic-go/quic-go/blob/v0.35.0/multiplexer.go#L59

### PR DESCRIPTION
**Pull request by Glonee**
Fixes #3868. Depends on #3867

### CODE REVIEW

**Comment by Glonee** [comment]
The integration test won't pass until #3867 is merged.

**Comment by marten-seemann** [comment]
I merged #3867. Could you rebase this PR?

File: http3/roundtrip.go [hunk 1]
```diff
@@ -18,7 +18,6 @@ import (
 )
 
 // declare this as a variable, such that we can it mock it in the tests
```
Comment (marten-seemann) [inline http3/roundtrip.go:20]:
```suggestion
```

File: integrationtests/self/http_test.go [hunk 2]
```diff
@@ -122,6 +123,21 @@ var _ = Describe("HTTP tests", func() {
 		Expect(string(body)).To(Equal("Hello, World!\n"))
 	})
 
+	It("request to server with two different servrname", func() {
```
Comment (marten-seemann) [inline integrationtests/self/http_test.go:126]: I'd like to explicitly check that this was using the same UDP conn. What do you think of introducing an HTTP handler on the server side that outputs the remote address, and then checking that both requests came from the same address?

Comment (Glonee) [reply integrationtests/self/http_test.go:126]: I will do it right away.

**Comment by Glonee** [comment]
Question: Does the `quic.Transport.Close()` close the underlying `UDPConn`? Otherwise we should close it manually in the `http3.RoundTripper.Close()`.

**Comment by marten-seemann** [comment]
> Question: Does the `quic.Transport.Close()` close the underlying `UDPConn`? Otherwise we should close it manually in the `http3.RoundTripper.Close()`.

It doesn't. Want to send another PR before I cut v0.35.1?

**Pull request merged**